docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,335 @@
+    or shared memory locations, the URI should include this parameter. This value is used to
+    tag messages sent from the client to the data server, containing specific offsets / addresses
+    which were provided that are no longer required by the client (i.e. any operations that
+    directly reference those memory locations, such as copying the remote data into local memory,
+    have been completed).
+* ``remote_handle`` - **OPTIONAL** - base64-encoded string
+  - When working with shared memory or remote memory, this value indicates any required
+    handle or identifier that is necessary for accessing the memory.
+    + Using UCX, this would be an *rkey* value
+    + With CUDA IPC, this would be the value of the base GPU pointer or memory handle,
+      and subsequent addresses would be offsets from this base pointer.
+
+Protocol Description
+====================
+
+There are two possibilities that can occur:
+
+1. The streams of metadata and body data are across separate connections
+
+.. figure:: ./DissociatedIPC/SequenceDiagramSeparate.mmd.svg
+
+2. The streams of metadata and body data are sent simultaneously across the
+   same connection
+
+.. figure:: ./DissociatedIPC/SequenceDiagramSame.mmd.svg
+
+Server Sequence
+---------------
+
+There can be either a single server handling both the IPC Metadata stream and the
+Body data streams, or separate servers for handling the IPC Metadata and the body
+data. This allows for streaming of data across either a single transport pipe or
+two pipes if desired.
+
+Metadata Stream Sequence
+''''''''''''''''''''''''
+
+The standing state of the server is waiting for a **tagged** message with the specified
+``<want_data>`` tag value to initiate a transfer. The body of that message will contain an
+opaque, binary identifier to indicate a particular dataset / data stream to send.
+
+.. note::
+
+  For instance, the **ticket** that was passed with a *FlightInfo* message would be
+  the body of this message. Because it is opaque, it can be anything the server wants
+  to use. The URI and identifier do not need to be given to the client via Flight RPC,
+  but could come across from any transport or protocol desired.
+
+Upon receiving a ``<want_data>`` request, the server *should* respond by sending a stream
+of messages consisting of the following:
+
+* A 5-byte prefix
+  - The first byte of the message indicates the type of message, currently there are only
+    two allowed message types (more types may get added in the future):
+    0) End of Stream
+    1) Flatbuffer IPC Metadata Message
+  - the next 4-bytes are a little-endian, 32-bit integer indicating the sequence number of
+    the message. The first message in the stream (**MUST** always be a schema message) **MUST**
+    have a sequence number of ``0``. Each subsequent message **MUST** increment the number by 
+    ``1``.
+* The full Flatbuffer bytes of an Arrow IPC header
```

Comment:
Flatbuffers

Reply:
Also, what would the Arrow IPC header represent exactly?

Reply:
the Flatbuffers Arrow IPC header is the metadata necessary for interpreting the buffers. i.e. the structure and nodes of a given message, along with the metadata of length, null_count, offset, etc. for each array etc.

Reply:
Since we link to the IPC encapsulation documentation earlier, I didn't feel it necessary to re-explain that here

Reply:
Hmm, ok. I was asking because the term "header" can be a bit under-defined.